File: Ahorn/entities/ObsidianShard.jl
module ObsidianShard

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/ObsidianShard" ObsidianShard(x::Integer, y::Integer)

const placements = Ahorn.PlacementDict(
    "normal" => Ahorn.EntityPlacement(ObsidianShard)
)

function Ahorn.selection(entity::ObsidianShard)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::ObsidianShard, room::Maple.Room)
    Ahorn.drawRectangle(ctx, Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16), (0.5, 0.5, 0.5, 0.5), (0.0, 0.0, 0.0, 1.0))
end

end

File: framework/extracted/Loenn/entities/obsidianShard.lua
local utils = require("utils")
local drawableSprite = require("structs.drawable_sprite")

local obsidianShard = {}

obsidianShard.name = "MaggyHelper/ObsidianShard"
obsidianShard.depth = -50
obsidianShard.placements = {
    {
        name = "normal",
        data = {}
    }
}

function obsidianShard.sprite(room, entity)
    local sprite = drawableSprite.fromTexture("objects/obsidian_shard/default", entity)
    sprite:setPosition(entity.x, entity.y)
    sprite:setJustification(0.5, 1.0)
    sprite.color = {0.2, 0.0, 0.3, 1.0}
    return sprite
end

function obsidianShard.selection(room, entity)
    return utils.rectangle(entity.x - 8, entity.y - 16, 16, 16)
end

return obsidianShard
